Source/WarpX.H

```diff
@@ -110,18 +110,18 @@ public:
     static std::string Version (); //!< Version of WarpX executable
     static std::string PicsarVersion (); //!< Version of PICSAR dependency
 
-    int Verbose () const { return verbose; }
+    int Verbose () const;
 
     void InitData ();
 
     void Evolve (int numsteps = -1);
 
-    MultiParticleContainer& GetPartContainer () { return *mypc; }
-    MacroscopicProperties& GetMacroscopicProperties () { return *m_macroscopic_properties; }
-    HybridPICModel& GetHybridPICModel () { return *m_hybrid_pic_model; }
-    MultiDiagnostics& GetMultiDiags () {return *multi_diags;}
+    MultiParticleContainer& GetPartContainer ();
+    MacroscopicProperties& GetMacroscopicProperties ();
+    HybridPICModel& GetHybridPICModel ();
+    MultiDiagnostics& GetMultiDiags ();
 
-    ParticleBoundaryBuffer& GetParticleBoundaryBuffer () { return *m_particle_boundary_buffer; }
```

Comment:
For these one-line definitions, it is indeed ok to have them in the header and also supports the compiler to inline them.

We can move them to cpp for stylistic reasons if we want, probably slightly increases link time.

Reply:
If I could give my opinion, I generally prefer one line routines like these to be in the header file. This reduces a lot of duplication of code that has little benefit. It also makes reading the code and tracing through the call chain easier since I quickly see the variable that is being returned. With the little routines in the .cpp file, there is now one more step in the call chain (and one more file) to look through just to find a variable name. This is something that I do often enough, that having this one more level of indirection will be annoying.